company-ycmd.el

```diff
@@ -222,16 +212,21 @@ the completion candidates.
 When `company-ycmd-enable-fuzzy-matching' is nil, check if
 candidate starts with PREFIX, whether to include candidate in
 candidates list."
-  (let ((completion-list (append completion-vector nil))
-        (candidates '()))
-    (dolist (candidate completion-list (nreverse candidates))
-      (when (or company-ycmd-enable-fuzzy-matching
-                (company-ycmd--prefix-candidate-p candidate prefix))
-        (let ((result (funcall construct-candidate-fn
-                               candidate prefix start-col)))
-          (if (listp result)
-              (setq candidates (append result candidates))
-            (push result candidates)))))))
+  (let* ((prefix-start-col (- (+ 1 (ycmd--column-in-bytes)) (length prefix)))
+         (prefix-size (- start-col prefix-start-col))
+         (prefix-str (substring-no-properties prefix 0 prefix-size))
+         candidates)
+    (dolist (candidate (append completion-vector nil) (nreverse candidates))
+      (let ((item (assoc 'insertion_text candidate)))
+        (when (s-present? prefix-str)
+          (setcdr item (concat prefix-str
+                               (substring-no-properties (cdr item)))))
```

Comment:
Only apply modification of insertion text if prefix differs from company and ycmd.